code/modules/mob/living/carbon/human/life.dm

```diff
@@ -35,8 +35,13 @@
 			dead_ticks ++
 			if(dead_ticks > TIME_BEFORE_DNR)
 				set_undefibbable()
-			else
-				med_hud_set_status()
+				
+			if(!client)
+				var/mob/dead/observer/ghost = get_ghost()
+				if(!ghost?.can_reenter_corpse)
+					set_undefibbable()
+
```

Comment:
this will set people that log out and back in while defibbable as undefibbable:
ghosts with no client will return null to get ghost

can_reenter_corpse is also already checked in get ghost and is not necessary